packages/network/src/bytes.js

```diff
@@ -1,23 +1,41 @@
+// @ts-check
 /// <reference path="./types.js" />
 import { encodeBase64, decodeBase64 } from '@endo/base64';
 
+/** @typedef {Data | Buffer | Uint8Array | Iterable<number>} SourceData */
+
+/**
+ * @param {SourceData} contents
+ */
+const coerceToByteArray = contents => {
+  if (typeof contents === 'string') {
+    contents = contents.split('').map(c => c.charCodeAt(0));
+  }
+  if (contents instanceof Uint8Array) {
+    // Reconstruct to ensure we have a Uint8Array and not a Buffer.
+    return new Uint8Array(
+      contents.buffer,
+      contents.byteOffset,
+      contents.byteLength,
+    );
+  }
+  return new Uint8Array(contents);
```

Comment:
Things like `str.split("").map()` are pretty bad in current XS, cf. https://github.com/endojs/endo/issues/1984
```suggestion
  if (typeof contents === 'string') {
    return Uint8Array.from(contents, c => {
      const b = c.charCodeAt(0);
      b <= 0xFF || Fail`character cannot be coerced to an octet: ${c}`;
      return b;
    });
  } else if (contents instanceof Uint8Array) {
    // Reconstruct to ensure we have a Uint8Array and not a Buffer.
    return new Uint8Array(
      contents.buffer,
      contents.byteOffset,
      contents.byteLength,
    );
  }
  return new Uint8Array(contents);
```